### ISSUE TITLE
Run away background processes

### ISSUE THREAD

**Issue by chrisb34** [dbeaver/dbeaver#18031]
### Description

Recent versions, I'm not sure since when, are chewing up system resources on my Mac, such that I need to close or, quite often, force quit, the application.

I have no idea what it is trying to do in the background.  I will be using the database explorer to look at local databases.

Version 22.2.3.202210231354
on MacOs Monterey 12.6 with 64Gb Ram

![image](https://user-images.githubusercontent.com/4489938/197788000-a231f967-0527-4629-8eea-3a468203c273.png)

### DBeaver Version

22.2.3.202210231354

### Operating System

acOs Monterey 12.6

### Database and driver

MySql ver 8.0.31-1.2.10-server
Local Client: MySql 5.7.21, com.mysql.cj.jdbc.Driver, mysql-connector-java-8.0.17

### Steps to reproduce

Open local database
wait ...

**Comment by flambchap** [comment]
Hi. Can you please try to generate a thread dump and attach it here? Here is the instruction: https://dbeaver.com/docs/wiki/Making-a-thread-dump/. Please make it when you encounter such behavior. Thank you.

**Comment by chrisb34** [comment]
Running at 594% of CPU !!

2022-10-28 09:53:59
Full thread dump OpenJDK 64-Bit Server VM (11.0.12+7 mixed mode):

Threads class SMR info:
_java_thread_list=0x000060000007d760, length=47, elements={
0x00007f87e800f000, 0x00007f87e502c000, 0x00007f87e5029800, 0x00007f87e7017000,
0x00007f87e7050800, 0x00007f87e680d800, 0x00007f87e784c800, 0x00007f87e784d800,
0x00007f87e807a800, 0x00007f87e7925800, 0x00007f87e82d4000, 0x00007f87e82d5000,
0x00007f87e733d000, 0x00007f87e7376000, 0x00007f87e7380800, 0x00007f87e82cc000,
0x00007f87e513f000, 0x00007f87e83fd800, 0x00007f87c5870800, 0x00007f87e6032000,
0x00007f87e75ce000, 0x00007f87e75f8000, 0x00007f87861c1000, 0x00007f87e75cc000,
0x00007f87c58e0000, 0x00007f87e794d800, 0x00007f87e608f800, 0x00007f87c59bf000,
0x00007f87c533f000, 0x00007f87e6834000, 0x00007f87e7495800, 0x00007f87e737d000,
0x00007f87e860b800, 0x00007f87e5300800, 0x00007f87e7f82000, 0x00007f87e7c83800,
0x00007f87e6841000, 0x00007f87e829d800, 0x00007f87e6873000, 0x00007f87e62f2000,
0x00007f87e61b4000, 0x00007f87e6088000, 0x00007f8786480800, 0x00007f87e8156000,
0x00007f87c5236800, 0x00007f87e6404000, 0x00007f87e6879000
}

"main" #1 prio=6 os_prio=31 cpu=1101431.00ms elapsed=2164.67s tid=0x00007f87e800f000 nid=0x103 runnable  [0x00007ff7b09f2000]
   java.lang.Thread.State: RUNNABLE
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lineLiesOnPreviouslyCreatedLine(MikamiTabuchiRouter.java:148)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createTrial(MikamiTabuchiRouter.java:109)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createLinesFromTrial(MikamiTabuchiRouter.java:94)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lambda$7(MikamiTabuchiRouter.java:451)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter$$Lambda$789/0x0000000100aa4040.accept(Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.solvePath(MikamiTabuchiRouter.java:451)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.updatePath(MikamiTabuchiRouter.java:319)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.solve(MikamiTabuchiRouter.java:282)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiConnectionRouter.lambda$2(MikamiTabuchiConnectionRouter.java:161)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiConnectionRouter$$Lambda$764/0x0000000100a5a840.run(Unknown Source)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
	- locked <0x00000000ce208820> (a org.eclipse.swt.widgets.RunnableLock)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4318)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3941)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:643)
	at org.eclipse.ui.internal.Workbench$$Lambda$208/0x000000010040e440.run(Unknown Source)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:550)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:171)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:292)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at ***@***.***/Native Method)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)

"Reference Handler" #2 daemon prio=10 os_prio=31 cpu=8.34ms elapsed=2164.65s tid=0x00007f87e502c000 nid=0x9a03 waiting on condition  [0x0000700009208000]
   java.lang.Thread.State: RUNNABLE
	at ***@***.***/Native Method)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"Finalizer" #3 daemon prio=8 os_prio=31 cpu=0.75ms elapsed=2164.65s tid=0x00007f87e5029800 nid=0x9703 in Object.wait()  [0x000070000930b000]
   java.lang.Thread.State: WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at ***@***.***/Unknown Source)
	- waiting to re-lock in wait() <0x00000000c00c9a80> (a java.lang.ref.ReferenceQueue$Lock)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"Signal Dispatcher" #4 daemon prio=9 os_prio=31 cpu=0.30ms elapsed=2164.64s tid=0x00007f87e7017000 nid=0x6a03 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Service Thread" #5 daemon prio=9 os_prio=31 cpu=0.05ms elapsed=2164.64s tid=0x00007f87e7050800 nid=0x9403 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #6 daemon prio=9 os_prio=31 cpu=22586.27ms elapsed=2164.64s tid=0x00007f87e680d800 nid=0x6e03 waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE
   No compile task

"C1 CompilerThread0" #9 daemon prio=9 os_prio=31 cpu=4178.09ms elapsed=2164.64s tid=0x00007f87e784c800 nid=0x7003 waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE
   No compile task

"Sweeper thread" #10 daemon prio=9 os_prio=31 cpu=31.42ms elapsed=2164.64s tid=0x00007f87e784d800 nid=0x7203 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Common-Cleaner" #11 daemon prio=8 os_prio=31 cpu=2.74ms elapsed=2164.53s tid=0x00007f87e807a800 nid=0x8e03 in Object.wait()  [0x000070000991d000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at ***@***.***/Unknown Source)
	- waiting to re-lock in wait() <0x00000000c03a3f48> (a java.lang.ref.ReferenceQueue$Lock)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"Active Thread: Equinox Container: c2164b61-1261-4159-8108-ec20b34366d3" #13 prio=5 os_prio=31 cpu=10.32ms elapsed=2163.53s tid=0x00007f87e7925800 nid=0xc503 waiting on condition  [0x000070000a33b000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at ***@***.***/Native Method)
	- parking to wait for  <0x00000000c06ac8a0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"Framework Event Dispatcher: Equinox Container: c2164b61-1261-4159-8108-ec20b34366d3" #15 daemon prio=5 os_prio=31 cpu=47.26ms elapsed=2163.34s tid=0x00007f87e82d4000 nid=0x14103 in Object.wait()  [0x000070000a644000]
   java.lang.Thread.State: WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at ***@***.***/Unknown Source)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
	- waiting to re-lock in wait() <0x00000000c06b6060> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:341)

"Start Level: Equinox Container: c2164b61-1261-4159-8108-ec20b34366d3" #16 daemon prio=5 os_prio=31 cpu=412.57ms elapsed=2163.33s tid=0x00007f87e82d5000 nid=0x14003 in Object.wait()  [0x000070000a747000]
   java.lang.Thread.State: WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <0x00000000c06acb10> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
	at ***@***.***/Unknown Source)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
	- waiting to re-lock in wait() <0x00000000c06acb10> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:341)

"SCR Component Actor" #17 daemon prio=5 os_prio=31 cpu=0.10ms elapsed=2163.02s tid=0x00007f87e733d000 nid=0x740b in Object.wait()  [0x0000700009b23000]
   java.lang.Thread.State: WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <0x00000000c06ace78> (a java.util.LinkedList)
	at ***@***.***/Unknown Source)
	at org.apache.felix.scr.impl.ComponentActorThread.run(ComponentActorThread.java:83)
	- waiting to re-lock in wait() <0x00000000c06ace78> (a java.util.LinkedList)
	at ***@***.***/Unknown Source)

"EMF Reference Cleaner" #19 daemon prio=5 os_prio=31 cpu=26.09ms elapsed=2162.03s tid=0x00007f87e7376000 nid=0x12d07 in Object.wait()  [0x000070000a84a000]
   java.lang.Thread.State: WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at ***@***.***/Unknown Source)
	- waiting to re-lock in wait() <0x00000000c06ad2e0> (a java.lang.ref.ReferenceQueue$Lock)
	at ***@***.***/Unknown Source)
	at org.eclipse.emf.common.util.CommonUtil$1ReferenceClearingQueuePollingThread.run(CommonUtil.java:70)

"Worker-JM" #20 prio=5 os_prio=31 cpu=0.12ms elapsed=2161.20s tid=0x00007f87e7380800 nid=0x12a03 in Object.wait()  [0x000070000aa50000]
   java.lang.Thread.State: WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <0x00000000c0aaf580> (a java.util.ArrayList)
	at org.eclipse.core.internal.jobs.InternalWorker.run(InternalWorker.java:61)
	- waiting to re-lock in wait() <0x00000000c0aaf580> (a java.util.ArrayList)

"Worker-0" #21 prio=5 os_prio=31 cpu=490.69ms elapsed=2161.06s tid=0x00007f87e82cc000 nid=0xdb03 in Object.wait()  [0x000070000ab53000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Database navigator events processor" #23 prio=5 os_prio=31 cpu=733.41ms elapsed=2159.60s tid=0x00007f87e513f000 nid=0x12307 waiting on condition  [0x000070000addf000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at ***@***.***/Native Method)
	at org.jkiss.dbeaver.utils.RuntimeUtils.pause(RuntimeUtils.java:166)
	at org.jkiss.dbeaver.model.navigator.DBNModel$EventProcessingJob.run(DBNModel.java:735)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

"RMI TCP Accept-37947" #25 daemon prio=6 os_prio=31 cpu=0.64ms elapsed=2159.16s tid=0x00007f87e83fd800 nid=0x11e03 runnable  [0x000070000b3f1000]
   java.lang.Thread.State: RUNNABLE
	at ***@***.***/Native Method)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"RMI TCP Accept-0" #26 daemon prio=6 os_prio=31 cpu=0.12ms elapsed=2159.15s tid=0x00007f87c5870800 nid=0x11c03 runnable  [0x000070000b4f4000]
   java.lang.Thread.State: RUNNABLE
	at ***@***.***/Native Method)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"RMI Reaper" #27 prio=6 os_prio=31 cpu=0.09ms elapsed=2159.15s tid=0x00007f87e6032000 nid=0x11a03 in Object.wait()  [0x000070000b5f7000]
   java.lang.Thread.State: WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <0x00000000c0dfec90> (a java.lang.ref.ReferenceQueue$Lock)
	at ***@***.***/Unknown Source)
	- waiting to re-lock in wait() <0x00000000c0dfec90> (a java.lang.ref.ReferenceQueue$Lock)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"RMI GC Daemon" #28 daemon prio=2 os_prio=31 cpu=0.37ms elapsed=2159.14s tid=0x00007f87e75ce000 nid=0xee03 in Object.wait()  [0x000070000b6fa000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <0x00000000c0de5918> (a sun.rmi.transport.GC$LatencyLock)
	at ***@***.***/Unknown Source)
	- waiting to re-lock in wait() <0x00000000c0de5918> (a sun.rmi.transport.GC$LatencyLock)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"Worker-2" #29 prio=5 os_prio=31 cpu=618.42ms elapsed=2159.05s tid=0x00007f87e75f8000 nid=0x11803 in Object.wait()  [0x000070000b7fd000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Bundle File Closer" #30 daemon prio=6 os_prio=31 cpu=15.74ms elapsed=2158.08s tid=0x00007f87861c1000 nid=0x11603 in Object.wait()  [0x000070000ba03000]
   java.lang.Thread.State: WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at ***@***.***/Unknown Source)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
	- waiting to re-lock in wait() <0x00000000c11af700> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:341)

"Worker-3" #31 prio=5 os_prio=31 cpu=471.94ms elapsed=2157.98s tid=0x00007f87e75cc000 nid=0xf503 in Object.wait()  [0x000070000bb06000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Worker-4" #33 prio=5 os_prio=31 cpu=737.45ms elapsed=2153.09s tid=0x00007f87c58e0000 nid=0x15827 in Object.wait()  [0x000070000b2ee000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"EventAdmin Async Event Dispatcher Thread" #34 daemon prio=6 os_prio=31 cpu=6.09ms elapsed=2139.51s tid=0x00007f87e794d800 nid=0x150f in Object.wait()  [0x00007000089ea000]
   java.lang.Thread.State: WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at ***@***.***/Unknown Source)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
	- waiting to re-lock in wait() <0x00000000c16df0d0> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:341)

"pool-1-thread-1" #35 prio=5 os_prio=31 cpu=6.99ms elapsed=2139.50s tid=0x00007f87e608f800 nid=0x12f0f waiting on condition  [0x000070000a541000]
   java.lang.Thread.State: WAITING (parking)
	at ***@***.***/Native Method)
	- parking to wait for  <0x00000000c16e3ea8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"pool-2-thread-1" #36 prio=5 os_prio=31 cpu=1.77ms elapsed=2139.50s tid=0x00007f87c59bf000 nid=0xdd0b waiting on condition  [0x000070000ac56000]
   java.lang.Thread.State: WAITING (parking)
	at ***@***.***/Native Method)
	- parking to wait for  <0x00000000c16e8b10> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"Worker-5" #38 prio=5 os_prio=31 cpu=505.11ms elapsed=2135.60s tid=0x00007f87c533f000 nid=0x1db03 in Object.wait()  [0x000070000c01b000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"mysql-cj-abandoned-connection-cleanup" #39 daemon prio=5 os_prio=31 cpu=38.62ms elapsed=2135.25s tid=0x00007f87e6834000 nid=0x1e713 in Object.wait()  [0x000070000be12000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at ***@***.***/Unknown Source)
	- waiting to re-lock in wait() <0x00000000c2222c10> (a java.lang.ref.ReferenceQueue$Lock)
	at com.mysql.cj.jdbc.AbandonedConnectionCleanupThread.run(AbandonedConnectionCleanupThread.java:85)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"Worker-6" #40 prio=5 os_prio=31 cpu=454.45ms elapsed=2134.39s tid=0x00007f87e7495800 nid=0x1d003 in Object.wait()  [0x000070000c324000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Worker-7" #43 prio=5 os_prio=31 cpu=624.58ms elapsed=2114.58s tid=0x00007f87e737d000 nid=0x1cd13 in Object.wait()  [0x0000700008aed000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Worker-8" #44 prio=5 os_prio=31 cpu=480.02ms elapsed=2074.27s tid=0x00007f87e860b800 nid=0x1705f in Object.wait()  [0x000070000bc09000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Worker-9" #45 prio=5 os_prio=31 cpu=255.89ms elapsed=1894.09s tid=0x00007f87e5300800 nid=0x1cc1f in Object.wait()  [0x000070000bd0c000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"ForkJoinPool.commonPool-worker-3" #47 daemon prio=6 os_prio=31 cpu=1235099.04ms elapsed=1857.11s tid=0x00007f87e7f82000 nid=0x19a07 runnable  [0x000070000c6b5000]
   java.lang.Thread.State: RUNNABLE
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lineLiesOnPreviouslyCreatedLine(MikamiTabuchiRouter.java:148)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createTrial(MikamiTabuchiRouter.java:110)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createLinesFromTrial(MikamiTabuchiRouter.java:86)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lambda$7(MikamiTabuchiRouter.java:451)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter$$Lambda$789/0x0000000100aa4040.accept(Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"ForkJoinPool.commonPool-worker-5" #48 daemon prio=6 os_prio=31 cpu=1231794.02ms elapsed=1857.11s tid=0x00007f87e7c83800 nid=0x1ae07 runnable  [0x000070000c7b8000]
   java.lang.Thread.State: RUNNABLE
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lineLiesOnPreviouslyCreatedLine(MikamiTabuchiRouter.java:144)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createTrial(MikamiTabuchiRouter.java:109)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createLinesFromTrial(MikamiTabuchiRouter.java:94)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lambda$7(MikamiTabuchiRouter.java:451)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter$$Lambda$789/0x0000000100aa4040.accept(Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"ForkJoinPool.commonPool-worker-7" #49 daemon prio=6 os_prio=31 cpu=1227119.47ms elapsed=1857.11s tid=0x00007f87e6841000 nid=0x1ac07 runnable  [0x000070000c8bb000]
   java.lang.Thread.State: RUNNABLE
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lineLiesOnPreviouslyCreatedLine(MikamiTabuchiRouter.java:148)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createTrial(MikamiTabuchiRouter.java:109)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createLinesFromTrial(MikamiTabuchiRouter.java:86)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lambda$7(MikamiTabuchiRouter.java:451)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter$$Lambda$789/0x0000000100aa4040.accept(Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"ForkJoinPool.commonPool-worker-9" #50 daemon prio=6 os_prio=31 cpu=1235277.47ms elapsed=1857.11s tid=0x00007f87e829d800 nid=0x1a907 runnable  [0x000070000c9be000]
   java.lang.Thread.State: RUNNABLE
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lineLiesOnPreviouslyCreatedLine(MikamiTabuchiRouter.java:148)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createTrial(MikamiTabuchiRouter.java:109)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createLinesFromTrial(MikamiTabuchiRouter.java:94)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lambda$7(MikamiTabuchiRouter.java:451)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter$$Lambda$789/0x0000000100aa4040.accept(Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"ForkJoinPool.commonPool-worker-11" #51 daemon prio=6 os_prio=31 cpu=1234858.68ms elapsed=1857.11s tid=0x00007f87e6873000 nid=0x19d07 runnable  [0x000070000cac1000]
   java.lang.Thread.State: RUNNABLE
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lineLiesOnPreviouslyCreatedLine(MikamiTabuchiRouter.java:144)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createTrial(MikamiTabuchiRouter.java:109)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createLinesFromTrial(MikamiTabuchiRouter.java:94)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lambda$7(MikamiTabuchiRouter.java:451)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter$$Lambda$789/0x0000000100aa4040.accept(Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"ForkJoinPool.commonPool-worker-15" #53 daemon prio=6 os_prio=31 cpu=1232173.52ms elapsed=1857.11s tid=0x00007f87e62f2000 nid=0x19f07 runnable  [0x000070000cbc4000]
   java.lang.Thread.State: RUNNABLE
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lineLiesOnPreviouslyCreatedLine(MikamiTabuchiRouter.java:148)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createTrial(MikamiTabuchiRouter.java:109)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createLinesFromTrial(MikamiTabuchiRouter.java:94)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lambda$7(MikamiTabuchiRouter.java:451)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter$$Lambda$789/0x0000000100aa4040.accept(Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"ForkJoinPool.commonPool-worker-13" #52 daemon prio=6 os_prio=31 cpu=1232219.90ms elapsed=1857.11s tid=0x00007f87e61b4000 nid=0x1a107 runnable  [0x000070000ccc7000]
   java.lang.Thread.State: RUNNABLE
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lineLiesOnPreviouslyCreatedLine(MikamiTabuchiRouter.java:148)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createTrialLine(MikamiTabuchiRouter.java:214)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createTrial(MikamiTabuchiRouter.java:103)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.createLinesFromTrial(MikamiTabuchiRouter.java:86)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter.lambda$7(MikamiTabuchiRouter.java:451)
	at org.jkiss.dbeaver.erd.ui.router.MikamiTabuchiRouter$$Lambda$789/0x0000000100aa4040.accept(Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)
	at ***@***.***/Unknown Source)

"Worker-10" #54 prio=5 os_prio=31 cpu=191.65ms elapsed=1576.28s tid=0x00007f87e6088000 nid=0x1941b in Object.wait()  [0x000070000bf15000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Worker-11" #55 prio=5 os_prio=31 cpu=97.61ms elapsed=926.52s tid=0x00007f8786480800 nid=0x1c14f in Object.wait()  [0x000070000c11e000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Worker-12" #56 prio=5 os_prio=31 cpu=93.59ms elapsed=829.97s tid=0x00007f87e8156000 nid=0x1e1af in Object.wait()  [0x000070000c221000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Worker-13" #57 prio=5 os_prio=31 cpu=64.17ms elapsed=638.14s tid=0x00007f87c5236800 nid=0x1e53b in Object.wait()  [0x000070000c427000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Worker-14" #58 prio=5 os_prio=31 cpu=26.15ms elapsed=265.41s tid=0x00007f87e6404000 nid=0x1ca3f in Object.wait()  [0x000070000c52a000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at ***@***.***/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
	- waiting to re-lock in wait() <0x00000000c0ab77f8> (a org.eclipse.core.internal.jobs.WorkerPool)
	at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Attach Listener" #59 daemon prio=9 os_prio=31 cpu=2.02ms elapsed=8.04s tid=0x00007f87e6879000 nid=0xae3f waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"VM Thread" os_prio=31 cpu=13885.27ms elapsed=2164.66s tid=0x00007f87e7016800 nid=0x9c03 runnable  

"GC Thread#0" os_prio=31 cpu=37037.03ms elapsed=2164.67s tid=0x00007f87e7827000 nid=0x6003 runnable  

"GC Thread#1" os_prio=31 cpu=36779.22ms elapsed=2163.55s tid=0x00007f87e5095000 nid=0x14803 runnable  

"GC Thread#2" os_prio=31 cpu=36831.49ms elapsed=2163.39s tid=0x00007f87c580b000 nid=0x14a07 runnable  

"GC Thread#3" os_prio=31 cpu=36757.24ms elapsed=2163.39s tid=0x00007f87c5877800 nid=0xbd07 runnable  

"GC Thread#4" os_prio=31 cpu=36876.69ms elapsed=2160.16s tid=0x00007f87e5380000 nid=0xe203 runnable  

"GC Thread#5" os_prio=31 cpu=37021.23ms elapsed=2160.16s tid=0x00007f87e51ca000 nid=0x12203 runnable  

"GC Thread#6" os_prio=31 cpu=36970.42ms elapsed=2160.16s tid=0x00007f87861c6000 nid=0xe403 runnable  

"GC Thread#7" os_prio=31 cpu=36895.83ms elapsed=2160.16s tid=0x00007f878627c800 nid=0x11f03 runnable  

"G1 Main Marker" os_prio=31 cpu=41.30ms elapsed=2164.67s tid=0x00007f87e7847000 nid=0xa303 runnable  

"G1 Conc#0" os_prio=31 cpu=3881.76ms elapsed=2164.67s tid=0x00007f87e700c800 nid=0xa003 runnable  

"G1 Conc#1" os_prio=31 cpu=3881.13ms elapsed=2161.27s tid=0x00007f87e6815800 nid=0x12c03 runnable  

"G1 Refine#0" os_prio=31 cpu=3.48ms elapsed=2164.67s tid=0x00007f87e8038000 nid=0x9f03 runnable  

"G1 Refine#1" os_prio=31 cpu=1.46ms elapsed=2163.54s tid=0x00007f87e692c800 nid=0x14703 runnable  

"G1 Refine#2" os_prio=31 cpu=1.01ms elapsed=2163.54s tid=0x00007f87e683a000 nid=0xc003 runnable  

"G1 Refine#3" os_prio=31 cpu=0.61ms elapsed=2163.54s tid=0x00007f87e8181000 nid=0x14503 runnable  

"G1 Refine#4" os_prio=31 cpu=0.37ms elapsed=2163.54s tid=0x00007f87e600c000 nid=0x14303 runnable  

"G1 Refine#5" os_prio=31 cpu=0.04ms elapsed=2163.54s tid=0x00007f87c50d8000 nid=0xc203 runnable  

"G1 Young RemSet Sampling" os_prio=31 cpu=881.83ms elapsed=2164.67s tid=0x00007f87e5026800 nid=0x6403 runnable  
"VM Periodic Task Thread" os_prio=31 cpu=973.62ms elapsed=2164.51s tid=0x00007f87e502f000 nid=0x8d03 waiting on condition  

JNI global refs: 1261, weak refs: 0

**Comment by Destrolaric** [comment]
@chrisb34 I see it's related to routing for diagrams. It looks like it can't handle the path creation for the ER diagram. Does your diagram have a lot of tables? You can change the routing method to the Shortest Path in ERD settings as a workaround.
![image](https://user-images.githubusercontent.com/20579256/199724696-ad323d96-36ad-4de5-8e39-7dbc86a30990.png)

**Comment by chrisb34** [comment]
Hi Georgii

Thanks for the reply. Yes that seems to work once I update that parameter and restart DBeaver.
Thank you!

**Comment by jacekgajek** [comment]
It is not related to the size of a schema, but a size of a viewport. I have a schema with 6 tables and 5 relationships.

Everything works fine if a window is maximized, but if I resize DBeaver window so the diagram doesn't fit then the CPU usage sky rockets.

**Comment by Destrolaric** [comment]
@jacekgajek Thank you for the information about how it could be reproduced

**Comment by jacekgajek** [comment]
How is it going? This bug is really frustrating, as it's easy to forget that a diagram is opened in some hidden tab and eats CPU.

**Comment by Destrolaric** [comment]
@jacekgajek, unfortunately still not fixed. The bug is related to Mikami-Tabuchi routing, you can set `Shortest path` as routing, and the issue will be not appear again.
![image](https://user-images.githubusercontent.com/20579256/217809313-9de7139f-f414-4000-9d4b-e3e69f85e0aa.png)

**Comment by mvoorberg** [comment]
Same high cpu experience with DBeaver 22.3.4 on Ubuntu. Switching back to `Shortest Path` solves it.

**Comment by amzbiz** [comment]
Have been getting it in DBeaver recently, but don't have any diagrams open so must be caused by something else?

(https://github.com/user-attachments/assets/64f9b75d-804b-4be5-bd3c-6957a3630b40)

**Comment by E1izabeth** [comment]
@amzbiz your case has nothing to do with this issue

**Comment by amzbiz** [comment]
Yes, you're right @E1izabeth, I suspect it's related to JSON in the result sets, but haven't managed to definitely confirm that yet. There's a few other similar issues, all seem close but not the same. It's quite annoying because it sends my CPU to 100% and basically makes DBeaver unusable. Then you have to kill it, which loses your file changes. I've looked at the logs, but they don't contain anything useful.

**Comment by E1izabeth** [comment]
Please make a thread dump, when it happen next time and create a new issue.

You can find instructions here: https://github.com/dbeaver/dbeaver/wiki/Making-a-thread-dump
